torch/ao/quantization/fx/utils.py

```diff
@@ -945,3 +953,50 @@ def find_patterns(
         new_input = first_tuple.args[0][last_getitem_index]  # type: ignore[index]
         for user in list(last_getitem.users.keys()):
             user.replace_input_with(last_getitem, new_input)
+
+def _maybe_validate_activation_post_process_dtype_constraints(
+        activation_post_process: Union[ObserverBase, FakeQuantize, None],
+        dtype_with_constraints: Optional[DTypeWithConstraints]):
+    """
+    Validate whether `activation_post_process` satisfies the following constraints from the backend,
+    specified through `dtype_with_constraints`:
+
+        1. Activation post process specified a quantization range that falls within the backend's, if any
+        2. Activation post process specified a min scale value that is >= the backend's, if any
+
+    If any of the above constraints are not satisfied, throw an error highlighting the offending values.
+
+    If either `activation_post_process` or `dtype_with_constraints` is None, or their dtypes do not match,
+    then do not validate.
+    """
+    if activation_post_process is None or \
+            dtype_with_constraints is None or \
+            activation_post_process.dtype != dtype_with_constraints.dtype:
+        return
+    debug_string = "Observer" if isinstance(activation_post_process, ObserverBase) else "FakeQuantize"
+    app_quant_min = getattr(activation_post_process, "quant_min", None)
+    app_quant_max = getattr(activation_post_process, "quant_max", None)
+    # TODO: for now, just use the existing eps value as scale_min. In the future, we should
+    # resolve the differences between the two, either by renaming eps or some other way
+    app_scale_min = getattr(activation_post_process, "eps", None)
+    backend_quant_min = dtype_with_constraints.quant_min_lower_bound
+    backend_quant_max = dtype_with_constraints.quant_max_upper_bound
+    backend_scale_min = dtype_with_constraints.scale_min_lower_bound
+    # validate quantization ranges
+    if backend_quant_min is not None and backend_quant_max is not None:
+        if app_quant_min is None or app_quant_max is None:
+            raise ValueError("%s must specify 'quant_min' and 'quant_max':\n%s" %
```

Comment:
the current behavior when something is not supported is that we just ignore the qconfig, should we error out here?

Reply:
maybe to keep consistent with the existing behavior we can put up a warning and ignore the qconfig?